sklearn/utils/extmath.py

```diff
@@ -523,13 +523,13 @@ def svd_flip(u, v, u_based_decision=True):
     if u_based_decision:
         # columns of u, rows of v
         max_abs_cols = np.argmax(np.abs(u), axis=0)
-        signs = np.sign(u[max_abs_cols, range(u.shape[1])])
+        signs = np.sign(u[max_abs_cols, list(range(u.shape[1]))])
```

Comment:
This was included because `cupy` arrays did not support slicing using `range`.

Reply:
Right, that is a real downside of passing the module to think about. I think it is a general downside of implicit dispatching possibly? That helper function could itself use `get_array_module()`. And if it uses `get_array_module()` that is totally fine, but means that other users have to be careful and possibly need transition.
Passing the module is as such maybe unnecessary, but seemed like an a safer transition start?

In many ways, I am starting to think that fully implicit may actually be nicest, but there are some downsides to figure out. The biggest one is the backward compatibility concern of array-providers though.

Reply:
Hmm, this comment should have been on the other one...

Reply:
These minor semantic incompatibilities will remain I think, in particular for indexing - I don't think there's a single array library that implements 100% of NumPy's indexing behavior. 

https://github.com/quansight/ndindex may help improve that situation.